package/gluon-web-admin/luasrc/lib/gluon/config-mode/model/admin/remote.lua

```diff
@@ -76,29 +75,12 @@ function pw2.cfgvalue()
 end
 
 local function set_password(password)
-	local inr, inw = unistd.pipe()
-	local pid = unistd.fork()
-
-	if pid < 0 then
-		return false
-	elseif pid == 0 then
-		unistd.close(inw)
-
-		local null = fcntl.open('/dev/null', fcntl.O_WRONLY)
-		unistd.dup2(null, unistd.STDOUT_FILENO)
-		unistd.dup2(null, unistd.STDERR_FILENO)
-		if null > 2 then
-			unistd.close(null)
-		end
-
-		unistd.dup2(inr, unistd.STDIN_FILENO)
-		unistd.close(inr)
-
-		unistd.execp('passwd', {[0] = 'passwd'})
-		os.exit(127)
-	end
+	local policies = {[unistd.STDIN_FILENO]=util.subprocess.PIPE,
+			  [unistd.STDOUT_FILENO]=util.subprocess.DEVNULL,
+			  [unistd.STDERR_FILENO]=util.subprocess.DEVNULL}
```

Comment:
@NeoRaider this would be less verbose for the calling function.

```suggestion
	local policies = {["stdin"]=util.subprocess.PIPE,
			  ["stdout"]=util.subprocess.DEVNULL,
			  ["stderr"]=util.subprocess.DEVNULL}
```

Reply:
I don't think its much cleaner; but creates the need for some kind of lookup in `popen()` to translate the stream names to file descriptors. Your choice.

Reply:
I just repasted the draft on how this change could reflect in the function: https://bpa.st/XHUQ (now with proper spacing)

Reply:
I'd expect the user to assign the `subprocess` module to a local variable:
```lua
sp.popen('ls', {'-l', '/'}, {
  stdin = sp.PIPE,
  stdout = sp.DEVNULL,
  stderr = sp.DEVNULL,
})
```

Reply:
resolved in 436b06d91094c3dbe49b8c40ec2c88e881cd87aa and e07c2abba3c51aef9af7e91055e04ff4516e9fd3